src/modules/dashboard/components/ExpenditurePage/Stages/ClaimFunds/ClaimFunds.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useMemo } from 'react';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+  useIntl,
+} from 'react-intl';
+import classNames from 'classnames';
+import { nanoid } from 'nanoid';
+
+import { FormSection } from '~core/Fields';
+import Tag from '~core/Tag';
+import TimeRelativeShort from '~dashboard/ExpenditurePage/TimeRelativeShort/TimeRelativeShort';
+import Button from '~core/Button';
+import Numeral from '~core/Numeral';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+
+import { buttonStyles } from '../Stages';
+import styles from './ClaimFunds.css';
+import { Recipient } from '~dashboard/ExpenditurePage/hooks';
+
+const displayName = 'dashboard.ExpenditurePage.ClaimFunds';
+
+const MSG = defineMessages({
+  claim: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claim',
+    defaultMessage: 'Next claim {claimDate}',
+  },
+  claimFunds: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimFunds',
+    defaultMessage: 'Claim funds',
+  },
+  totalClaimable: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.totalClaimable',
+    defaultMessage: 'Total claimable',
+  },
+  claimableNow: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimableNow',
+    defaultMessage: 'Claimable now',
+  },
+  claimed: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.claimed',
+    defaultMessage: 'Claimed',
+  },
+  nothingToClaim: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.nothingToClaim',
+    defaultMessage: 'Nothing to claim',
+  },
+  now: {
+    id: 'dashboard.ExpenditurePage.Stages.ClaimFunds.now',
+    defaultMessage: 'now',
+  },
+});
+
+export type Token = Record<string, number>;
+
+interface TokenWithId {
+  symbol: string;
+  amount: number;
+  id: string;
+}
+
+interface Props {
+  buttonAction?: () => void;
+  buttonText?: string | MessageDescriptor;
+  buttonIsActive: boolean;
+  nextClaimableRecipient?: Recipient;
+  totalClaimable?: Token;
+  claimableNow?: Token;
+  claimed: Token;
+}
+
+const ClaimFunds = ({
+  buttonAction,
+  buttonText,
+  buttonIsActive,
+  nextClaimableRecipient,
+  totalClaimable,
+  claimableNow,
+  claimed,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const convertToTokensWithIds = useMemo(
```

Comment:
I think this is unnecessary. You can just use nanoid() as key when you call numeral.